client/update.go

```diff
@@ -276,7 +279,7 @@ func (c *Channel) handleUpdateReq(
 		return
 	}
 
-	uh.HandleUpdate(req.ChannelUpdate, responder)
+	uh.HandleUpdate(c.machine.State(), req.ChannelUpdate, responder)
```

Comment:
I think we should not provide a reference to the actual state, but only a reference to a clone: `c.machine.State().Clone()`, right?

Reply:
`Channel.State` also returns a pointer to the machine state, but i can clone it if you want.